Add '-x' flag which to extract code

josephsnyder opened the pull request:
Add a '-x' flag which will utilize the ViViaN generation process but will
leave the execution after the export of the routines from the VistA
instance.  This directory can then be copied out of a container after it
has been initalized.

josephsnyder committed:
Add '-x' flag which to extract code

Add a '-x' flag which will utilize the ViViaN generation process but will
leave the execution after the export of the routines from the VistA
instance.  This directory can then be copied out of a container after it
has been initalized.

ChristopherEdwards commented on ViViaN/vivianInstall.sh:
```diff
@@ -13,7 +13,7 @@ usage()
 }
 
-while getopts ":hi:s" option
+while getopts ":h:x:i:s:" option
```
Why does extracting need another argument? aka -x somethingHere. It wasn't used on the main script.

ChristopherEdwards commented on README.md:
```diff
@@ -104,6 +104,22 @@ Example:
 
     docker build --build-arg flags="-y -b -s -a https://github.com/OSEHRA/vxVistA-M/archive/master.zip" -t vxvista .
 
+##### available flags
```
not sure that all autoInstaller flags should be here. It is just one more place that needs to be updated if more options are added in the future. Help is generated by the autoinstaller script. maybe just reference that section.

Though the specific option for doing extraction should be called out in this document - How to extract M components from a CACHE.DAT file.

shabiel commented:
Joe Snyder,

Should I wait for you to make updates based on Chris' feedback or start
testing it now?

--Sam

On Wed, May 16, 2018 at 4:43 PM, Christopher Edwards <
notifications@github.com> wrote:

> *@ChristopherEdwards* commented on this pull request.
> ------------------------------
>
> In README.md
> <https://github.com/OSEHRA/docker-vista/pull/5#discussion_r188766181>:
>
> > @@ -104,6 +104,22 @@ Example:
>
>      docker build --build-arg flags="-y -b -s -a https://github.com/OSEHRA/vxVistA-M/archive/master.zip" -t vxvista .
>
> +##### available flags
>
> not sure that all autoInstaller flags should be here. It is just one more
> place that needs to be updated if more options are added in the future.
> Help is generated by the autoinstaller script. maybe just reference that
> section.
>
> Though the specific option for doing extraction should be called out in
> this document - How to extract M components from a CACHE.DAT file.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/OSEHRA/docker-vista/pull/5#pullrequestreview-120812036>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABEtwosvuWzkcGyl3DADgcqBRTFtdcezks5tzI9jgaJpZM4UB-mA>
> .
>

josephsnyder commented:
I'll be making fixes soon.  Please hold off until then.

- Joe

Joseph Snyder
R&D Engineer

Kitware, Inc.
28 Corporate Dr
Clifton Park, NY 12065
(518) 881-4409

On 5/17/2018 9:38 AM, Sam Habiel wrote:
> Joe Snyder,
>
> Should I wait for you to make updates based on Chris' feedback or start
> testing it now?
>
> --Sam
>
> On Wed, May 16, 2018 at 4:43 PM, Christopher Edwards <
> notifications@github.com> wrote:
>
> > *@ChristopherEdwards* commented on this pull request.
> > ------------------------------
> >
> > In README.md
> > <https://github.com/OSEHRA/docker-vista/pull/5#discussion_r188766181>:
> >
> > > @@ -104,6 +104,22 @@ Example:
> >
> > docker build --build-arg flags="-y -b -s -a 
> https://github.com/OSEHRA/vxVistA-M/archive/master.zip" -t vxvista .
> >
> > +##### available flags
> >
> > not sure that all autoInstaller flags should be here. It is just one 
> more
> > place that needs to be updated if more options are added in the future.
> > Help is generated by the autoinstaller script. maybe just reference that
> > section.
> >
> > Though the specific option for doing extraction should be called out in
> > this document - How to extract M components from a CACHE.DAT file.
> >
> > —
> > You are receiving this because your review was requested.
> > Reply to this email directly, view it on GitHub
> > 
> <https://github.com/OSEHRA/docker-vista/pull/5#pullrequestreview-120812036>,
> > or mute the thread
> > 
> <https://github.com/notifications/unsubscribe-auth/ABEtwosvuWzkcGyl3DADgcqBRTFtdcezks5tzI9jgaJpZM4UB-mA>
> > .
> >
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/OSEHRA/docker-vista/pull/5#issuecomment-389869334>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe-auth/AAhFmv-dVG_P2y9rZiGZnNh_et2ydlMJks5tzX1wgaJpZM4UB-mA>.
>

josephsnyder committed:
Update from Review comments

Switch to using an argument-less flag for the pass of the extractOnly
information to vivianInstall.sh.

Remove the listing of flags and add a message to execute the
autoinstaller to see the list of flags.

Add example for export only run of instance.

ChristopherEdwards commented:
You may want to use a volume to mount the cache-files directory instead of a copy into the container. then the files will be on your file system and you can then use the docker --rm flag.

josephsnyder committed:
Add updates to foia Post install

Add comments and an additional piece needed to stop the "LOG RESOURCE
USAGE" from being executed by default, which causes errors when built on
a GT.M instance

josephsnyder commented:
Hi @ChristopherEdwards and @shabiel ,

The recent updates are just a rebase after the merging of the GT.M enhancements.  PTAL again to make sure your approval still stands.  Thanks!

ChristopherEdwards commented:
Approval still stands after rebase.
